packages/amplify-cli/src/extensions/amplify-helpers/delete-project.js

```diff
@@ -2,14 +2,27 @@ const ora = require('ora');
 const pathManager = require('./path-manager');
 const { removeEnvFromCloud } = require('./remove-env-from-cloud');
 const { getFrontendPlugins } = require('./get-frontend-plugins');
-
+const { getPluginInstance } = require('./get-plugin-instance');
+const { getAmplifyAppId } = require('./get-amplify-appId');
 async function deleteProject(context) {
   const confirmation = await getConfirmation(context);
   if (confirmation.proceed) {
     const allEnvs = context.amplify.getEnvDetails();
     const spinner = ora('Deleting resources from the cloud. This may take a few minutes...');
-    spinner.start();
-    await Promise.all(Object.keys(allEnvs).map(env => removeEnvFromCloud(context, env, confirmation.deleteS3)));
+
+    try {
+      spinner.start();
+      await Promise.all(Object.keys(allEnvs).map(env => removeEnvFromCloud(context, env, confirmation.deleteS3)));
+      const appId = getAmplifyAppId();
+      if (confirmation.deleteAmpilfyApp && appId) {
```

Comment:
this still has the risk of client un-intentionally deleting a Console project that contains env from other project.

Reply:
Fixed